riscv_ctg/env/arch_test.h

```diff
@@ -583,19 +583,44 @@
   .set offset,offset+REGWIDTH;\
   .endif;
   
-#define RVTEST_SIGUPD_F(_BR,_R,_F,...)\
-  .if NARG(__VA_ARGS__) == 1;\
-    FSREG _R,_ARG1(__VA_ARGS__,0)(_BR);\
-    SREG _F,_ARG1(__VA_ARGS__,0)+REGWIDTH(_BR);\
-    .set offset,_ARG1(__VA_OPT__(__VA_ARGS__,)0)+(REGWIDTH+REGWIDTH);\
-  .endif;\
-  .if NARG(__VA_ARGS__) == 0;\
-    FSREG _R,offset(_BR);\
-    SREG _F,offset+REGWIDTH(_BR);\
-    .set offset,offset+(REGWIDTH+REGWIDTH);\
-  .endif;
+#define RVTEST_SIGUPD_F(_BR,_R,_F,...) \
+  FLENN();\
```

Comment:
This does not solve the problems. The `FLENN` macro is still redefining the `LREG` and `SREG` macros which are used for the base extension tests.

Reply:
So, should I need to use the FLENN() macro in each of the test macros which is used in F tests like inside FPRR_OP  and FPIO_OP ?

Reply:
The FLENN() macro will redefine the LREG and SREG only for F test alone. As it is called only inside the macros of f tests.